node/src/components/storage/tests.rs

```diff
@@ -1152,3 +1181,133 @@ fn should_hard_reset() {
     // Test with a hard reset to era 0, deleting all blocks and associated data.
     check(0);
 }
+
+#[derive(Debug, Clone, PartialEq)]
+struct DatabaseEntriesSnapshot {
+    block_body_keys: HashSet<Digest>,
+    deploy_hashes_keys: HashSet<Digest>,
+    transfer_hashes_keys: HashSet<Digest>,
+    proposer_keys: HashSet<Digest>,
+}
+
+impl DatabaseEntriesSnapshot {
+    fn from_storage(storage: &Storage) -> DatabaseEntriesSnapshot {
+        let txn = storage.env.begin_ro_txn().unwrap();
+
+        let mut cursor = txn.open_ro_cursor(storage.block_body_v2_db).unwrap();
+        let block_body_keys = cursor
+            .iter()
+            .map(|(raw_key, _)| Digest::try_from(raw_key).unwrap())
+            .collect();
+        drop(cursor); // borrow checker complains without this
+
+        let mut cursor = txn.open_ro_cursor(storage.deploy_hashes_db).unwrap();
+        let deploy_hashes_keys = cursor
+            .iter()
+            .map(|(raw_key, _)| Digest::try_from(raw_key).unwrap())
+            .collect();
+        drop(cursor); // borrow checker complains without this
+
+        let mut cursor = txn.open_ro_cursor(storage.transfer_hashes_db).unwrap();
+        let transfer_hashes_keys = cursor
+            .iter()
+            .map(|(raw_key, _)| Digest::try_from(raw_key).unwrap())
+            .collect();
+        drop(cursor); // borrow checker complains without this
+
+        let mut cursor = txn.open_ro_cursor(storage.proposer_db).unwrap();
+        let proposer_keys = cursor
+            .iter()
+            .map(|(raw_key, _)| Digest::try_from(raw_key).unwrap())
+            .collect();
+        drop(cursor); // borrow checker complains without this
+
+        txn.commit().unwrap();
+
+        DatabaseEntriesSnapshot {
+            block_body_keys,
+            deploy_hashes_keys,
+            transfer_hashes_keys,
+            proposer_keys,
+        }
+    }
+}
+
+#[test]
+fn should_garbage_collect() {
+    let blocks_count = 9_usize;
```

Comment:
The comment below says 8 blocks?

```suggestion
    let blocks_count = 8_usize;
```

Reply:
Ah - I need to update the comment, then. I deliberately made it 9, so that the chain has a switch block at the end and one more snapshot will be taken.